src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -31,73 +30,70 @@ function getWallet(provider?: JsonRpcProvider | Eip1193Provider) {
   }
 }
 
-function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: string | JsonRpcProvider) {
-  // WalletConnect relies on Buffer, so it must be polyfilled.
-  if (!('Buffer' in window)) {
-    window.Buffer = Buffer
-  }
-
-  // FIXME(kristiehuang): we don't know what the props.jsonRpcEndpoint chain is; assume mainnet for WC instantiation
-  let mainnetRpcUrl: string | undefined
-  if (JsonRpcProvider.isProvider(jsonRpcEndpoint)) {
-    mainnetRpcUrl = jsonRpcEndpoint.connection.url
-  } else {
-    mainnetRpcUrl = jsonRpcEndpoint
-  }
+function getWalletConnectConnection(
+  useDefault: boolean,
+  jsonRpcUrlMap: { [chainId: number]: string[] },
+  defaultChainId: number
+) {
   return toWeb3Connection(
     initializeConnector<WalletConnect>(
       (actions) =>
         new WalletConnect({
           actions,
           options: {
-            rpc: {
-              [SupportedChainId.MAINNET]: [
-                mainnetRpcUrl ?? '',
-                ...(JSON_RPC_FALLBACK_ENDPOINTS[SupportedChainId.MAINNET] ?? []),
-              ].filter((url) => url !== undefined && url !== ''),
-              [SupportedChainId.RINKEBY]: JSON_RPC_FALLBACK_ENDPOINTS[SupportedChainId.RINKEBY] ?? [],
-            },
+            rpc: jsonRpcUrlMap,
             qrcode: useDefault,
           },
+          defaultChainId,
         })
     )
   )
 }
 
 interface ActiveWeb3ProviderProps {
   provider?: Eip1193Provider | JsonRpcProvider
-  jsonRpcEndpoint?: string | JsonRpcProvider
+  jsonRpcUrlMap: { [chainId: number]: string[] }
+  defaultChainId: number
 }
 
 export function ActiveWeb3Provider({
   provider: propsProvider,
-  jsonRpcEndpoint,
+  jsonRpcUrlMap,
+  defaultChainId: propsDefaultChainId,
   children,
 }: PropsWithChildren<ActiveWeb3ProviderProps>) {
+  if (propsDefaultChainId) defaultChainId = propsDefaultChainId
```

Comment:
okay so now that `defaultChainId: number` is required, i think you can nuke the local `export let defaultChainId = 1` and `propsDefaultChainId` and just use whatever you get from the parent, right?